src/H5Centry.c

```diff
@@ -1007,7 +1007,8 @@ H5C__load_entry(H5F_t *f,
     /* Call the get_initial_load_size callback, to retrieve the initial size of image */
     if (type->get_initial_load_size(udata, &len) < 0)
         HGOTO_ERROR(H5E_CACHE, H5E_CANTGET, NULL, "can't retrieve image size");
-    assert(len > 0);
+    if (len == 0)
```

Comment:
Seems to me this should really be checked in the callbacks, and left as an assert here.

Reply:
Do you know why the callback is returning len=0 wihtout returning an error?